### DISCUSSION TITLE
Question: `CanvasControl` not updating unless Window is resized. How do I make it update like Win2D `CanvasAnimatedControl`?

### DISCUSSION THREAD

**Discussion by myokeeh** [microsoft/microsoft-ui-xaml#7411] [Q&A]
I just did my preliminary "upgrade" using upgrade-assistant of a Kiosk UWP with Win2D. It's now in a state that I can run it and most of the code seems to work.

I found another thread here that says `CanvasAnimatedControl` can be replaced by `CanvasControl`, however the app now doesn't show what's drawn unless I resize the app window. 

As the name implies, maybe that's what sets `CanvasAnimatedControl` apart from `CanvasControl`. Is it something that hasn't been ported over?

**Comment by DHancock** [comment]
Unlike a CanvasAnimatedControl, the CanvasControl doesn't automatically redraw. You need to call [CanvasControl.Invalidate()](https://microsoft.github.io/Win2D/WinUI3/html/M_Microsoft_Graphics_Canvas_UI_Xaml_CanvasControl_Invalidate.htm) when it's content has changed.

**Comment by myokeeh** [comment]
@DHancock that's adequate for what I'm doing since I'm not really animating anything--displaying something new every half minute.

**Comment by michael-hawker** [comment]
@myokeeh can you call `CanvasControl.Invalidate` then on the SizeChanged event for your parent layout/window?

**Comment by myokeeh** [comment]
@michael-hawker, I'm doing that every time I draw something now. This particular app is supposed to be a full screen Kiosk app, so it is not allowed to resize. On to tackling getting rid of the title bar and bugging someone about lack of Kiosk mode support with WinUI/WindowsAppSDK.

**Comment by michael-hawker** [comment]
Oh, sorry I inverted the statement you mentioned about it only drawing when resizing the app vs. not drawing when resizing the app.
